pkg/types/chain_writer.go

```diff
@@ -27,6 +27,7 @@ type ChainWriter interface {
 type TxMeta struct {
 	// Used for Keystone Workflows
 	WorkflowExecutionID *string
+	GasLimit            *big.Int
```

Comment:
Let's document this well. In particular, must every chain implement this feature? If not, what happens when it is not supported?

Reply:
sorted